File: C#/SecuJournal/sjName.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace SecuJournal
{
    public partial class sjName : Form
    {
        public sjName()
        {
            InitializeComponent();
        }

        private void OKButton_Click(object sender, EventArgs e)
        {
            var mdiParent = Application.OpenForms.OfType<sjMain>().FirstOrDefault();
            sjMain sjm = new sjMain();
            sjEntry sje = new sjEntry();
            sje.MdiParent = mdiParent;
            sje.Show();

            if (TextBox1.Text == "")
            {
                sje.TextBox1.Text = "New journal entry - " + DateTime.Now.ToString("dd-MM-yyyy");
            }
            else
            {
                sje.TextBox1.Text = TextBox1.Text;
                sje.DateTimePicker1.Text = DateTimePicker1.Text;
            }

            if (Properties.Settings.Default.WriteTitle == true)
            {

            }
            else
            {
                sje.RichTextBox1.Text = sje.TextBox1.Text + " - " + sje.DateTimePicker1.Text;
            }
            this.Close();
        }

        private void CancelButton_Click(object sender, EventArgs e)
        {
            this.Close();
        }
    }
}

File: SecuJournal/SJName.vb
Public Class SJName
    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles OKButton.Click
        Dim NewEntry As New SJEntry
        NewEntry.MdiParent = SJMain
        NewEntry.Show()

        If TextBox1.Text = "" Then
            NewEntry.TextBox1.Text = "New journal entry - " + Date.Now.ToString("dd-MM-yyyy")
        Else
            NewEntry.TextBox1.Text = TextBox1.Text
            NewEntry.DateTimePicker1.Text = DateTimePicker1.Text
        End If


        'Load write title settings.
        If My.Settings.WriteTitle = 0 Then

        Else
            NewEntry.RichTextBox1.Text = NewEntry.TextBox1.Text + " - " + NewEntry.DateTimePicker1.Text
        End If

        Me.Close()
    End Sub

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles CancelButton.Click
        Me.Close()
    End Sub
End Class